### PR TITLE
ci(builder): Add CI for AutoGPT Builder

### PR DESCRIPTION
**Pull request by aarushik93**
### Background
Adding a CI workflow. For now just linting, because we don't have much else.

### Changes 🏗️
Created a autogpt-builder-ci.yml file, which runs the linting for our builder app

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
enhancement, configuration changes

___

### **Description**
- Added a new GitHub Actions workflow file `autogpt-builder-ci.yml` to set up continuous integration for the AutoGPT builder.
- Configured the workflow to trigger on pushes and pull requests to the main branch, specifically for changes in the `rnd/autogpt_builder` directory.
- Included steps in the workflow to check out the code, set up Node.js, install dependencies, and run linting.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>autogpt-builder-ci.yml</strong><dd><code>Add CI workflow for AutoGPT builder with linting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
.github/workflows/autogpt-builder-ci.yml

<li>Added a new GitHub Actions workflow for CI.<br> <li> Configured the workflow to run on pushes and pull requests to the main <br>branch.<br> <li> Included steps to set up Node.js, install dependencies, and run <br>linting.<br>

</details>
    

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7267/files#diff-86602f82a62584cf64806590cdfc82d634189b5065cca072466e276fa03e2113">+30/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: .github/workflows/autogpt-builder-ci.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: AutoGPT Builder CI
+
+on:
+  push:
+    branches: [ main ]
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: `master`

File: .github/workflows/autogpt-builder-ci.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    paths:
+      - 'rnd/autogpt_builder/**'
+  pull_request:
+    branches: [ main ]
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: CI for pull requests to feature branches could be nice too

File: .github/workflows/autogpt-builder-ci.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Node.js
+      uses: actions/setup-node@v3
+      with:
+        node-version: '16'
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: Node v16 is EOL as of September last year

File: .github/workflows/autogpt-builder-ci.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Install dependencies
+      run: |
+        cd rnd/autogpt_builder
+        npm install
+    - name: Run lint
+      run: |
+        cd rnd/autogpt_builder
+        npm run lint
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: missing newline

File: .github/workflows/autogpt-builder-ci.yml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    branches: [ main ]
+    paths:
+      - 'rnd/autogpt_builder/**'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]:
```suggestion
    - uses: actions/checkout@v4
```

File: .github/workflows/autogpt-builder-ci.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Set up Node.js
+      uses: actions/setup-node@v3
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]:
```suggestion
      uses: actions/setup-node@v4
```

File: .github/workflows/autogpt-builder-ci.yml [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      with:
+        node-version: '16'
+    - name: Install dependencies
+      run: |
+        cd rnd/autogpt_builder
+        npm install
+    - name: Run lint
+      run: |
+        cd rnd/autogpt_builder
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: You can avoid the repeated `cd` like this:
https://github.com/Significant-Gravitas/AutoGPT/blob/6e1c9d44a4db87d508059eee180788519475637a/.github/workflows/autogpt-ci.yml#L21-24

**Review by Swiftyos — APPROVED** [review]

**Review by Pwuts — DISMISSED** [review]

File: .github/workflows/autogpt-builder-ci.yml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+name: AutoGPT Builder CI
+
+on:
+  push:
+    branches: [ master ]
+    paths:
+      - 'rnd/autogpt_builder/**'
+  pull_request:
+    paths:
+      - 'rnd/autogpt_builder/**'
```
Comment (Pwuts) [inline .github/workflows/autogpt-builder-ci.yml]: Sorry for not mentioning this earlier, but the trigger filter should probably also include this workflow file:
```suggestion
    paths:
      - '.github/workflows/autogpt-builder-ci.yml'
      - 'rnd/autogpt_builder/**'
  pull_request:
    paths:
      - '.github/workflows/autogpt-builder-ci.yml'
      - 'rnd/autogpt_builder/**'
```

**Review by Pwuts — APPROVED** [review]

**Pull request merged**